src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ELEMENTS = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_ELEMENTS];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (array.length == size) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (array.length == size) {
+            grow();
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (array.length < size + list.size()) {
+            grow();
+        }
+        addition(list);
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size) {
+            return array[index];
+        } else {
+            throw new ArrayIndexOutOfBoundsException("Incorrect index");
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size) {
+            array[index] = value;
+        } else {
+            throw new ArrayIndexOutOfBoundsException("Incorrect index");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T tmp = array[index];
+        System.arraycopy(array, index + 1, array, index, size - index - 1);
+        size--;
+        return tmp;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (t == null) {
+                size--;
+                return null;
+            }
+            if (array[i].equals(t)) {
+                return remove(i);
+            }
```

Comment:
we have already compared two objects considering nulls multiple times in previous tasks
let's recall this)